vtex/omnishipping
# CHK-347: Fix address not updated with falsy postal code value

#### What is the purpose of this pull request?

Replaces the check for postal code `value` with `valid` flag in order to update the shipping data with the autocompleted address and avoid showing unavailable delivery options for (valid) addresses without postal code.

#### What problem is this solving?

In argentinian stores with geolocation active, if you searched an address that didn't had a postal code value returned from the Google API the UI would only show a warning saying that the inputted address didn't have any delivery options (the SLA array was empty and the address had all the required fields filled). But, if you made an action that would trigger Omnishipping to update the shipping data attachment with that autocompleted address (e.g. change the tab from delivery to pickup), it would suddenly have all delivery options correctly returned to it and the customer would be able to proceed the purchase.

This happens because one of the condition Omnishipping uses to check whether or not it will update the shipping data with the new address looked only at the existence of the postal code value, instead of if it was valid or not, which caused it to not even update the shipping data with this address-without-zipcode and display the message that shipping is unavailable. With these changes, it will now update the shipping data with this address before showing the unavailable message, and will only show it if the address really don't have any SLA to attend.

#### How should this be manually tested?

[Workspace](http://lucas--poccarrefourarg.myvtex.com/checkout/cart/add/?sku=144&qty=1&seller=1&sc=1).

You can test with the following addresses:

- av maipu 3886 (doesn't have postal code returned from google).
- av maipu 1860 (has zipcode).
- av del libertador 1700 (doesn't have neither zipcode nor city).

You might want to run the following snippet in the console to clear the addresses between each attempt to start from the same clean state

```js
vtexjs.checkout.sendAttachment('shippingData', { address: null, availableAddresses: [], selectedAddresses: [] })
```

You can see the current behaviour in [master](https://poccarrefourarg.myvtex.com/checkout/cart/add/?sku=144&qty=1&seller=1&sc=1).

#### Screenshots or example usage

#### Types of changes
